File: insertion_sort/sort.py
from timeit import default_timer as timer

def get_words():
  fname = "../dataset/words_10000.txt"

  wordlist = []

  with open(fname, "r") as fd:
    for line in fd:
      wordlist.append(line.strip())

  return wordlist

# O(n^2) - Best case O(n)
def insertion_sort(wordlist):
  for i in range(1, len(wordlist)):
    key = wordlist[i]
    j = i - 1

    while (j >= 0) and (wordlist[j] > key):
      wordlist[j+1] = wordlist[j]
      j = j - 1

    wordlist[j + 1] = key

wordlist = get_words()
wordlist2 = get_words()

print(wordlist[:10], wordlist[-10:])

tsort = timer()
insertion_sort(wordlist)
tsort = timer() - tsort

tsort2 = timer()
wordlist2.sort()
tsort2 = timer() - tsort2

print(wordlist[:10], wordlist[-10:])
print(wordlist2[:10], wordlist2[-10:])

print(str(tsort), str(tsort2))

File: selection_sort/sort.dart
import 'dart:io';
import 'dart:core';

List<String> get_words() {
  String fname = "../dataset/words_10000.txt";
  File fd = new File(fname);
  return fd.readAsStringSync().split("\n").where((el) => el.isNotEmpty).toList();
}

void selection_sort(List<String> list) {
  for (int i = 0; i < list.length; i++) {
    for (int j = i+1; j < list.length; j++) {
      if (list[i].compareTo(list[j]) > 0) {
        var aux = list[i];
        list[i] = list[j];
        list[j] = aux;
      }
    }
  }
}

void main() {
  List wordlist = get_words();
  List wordlist2 = get_words();

  print(wordlist);
  Stopwatch timer1 = new Stopwatch();
  Stopwatch timer2 = new Stopwatch();

  timer1.start();
  selection_sort(wordlist);
  timer1.stop();

  timer2.start();
  wordlist2.sort();
  timer2.stop();

  print(wordlist);
  print(wordlist2);
  print("${timer1.elapsedMicroseconds}, ${timer2.elapsedMicroseconds}");
}